src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -392,13 +392,18 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 					}
 				}
 				return this._fileBasedRecommendations[a].recommendedTime > this._fileBasedRecommendations[b].recommendedTime ? -1 : 1;
-			})
+			});
+		return [...fileBased, ...this.getRestoredRecommendationsByReason(ExtensionRecommendationReason.File)]
```

Comment:
we will need to store the sources as well, otherwise the `this._fileBasedRecommendations[extensionId]` would show up as undefined for restored recommendations.

Let's simplify this by just removing the extension from the ignored list instead of trying to restore them in real time when user chooses to undo 

The msg can be `You will receive recommendations for this extension in your next VS Code session`